src/components/CSSTable.tsx

```diff
@@ -0,0 +1,268 @@
+import { debounce, List, ListItem, Stack } from '@patternfly/react-core'
+import { Table, Thead, Th, Tr, Tbody, Td } from '@patternfly/react-table'
+import { useState } from 'react'
+import LevelUpAltIcon from '@patternfly/react-icons/dist/esm/icons/level-up-alt-icon'
+import * as tokensModule from '@patternfly/react-tokens/dist/esm/componentIndex'
+import React from 'react'
+import { CSSSearch } from './CSSSearch'
+import { AutoLinkHeader } from './AutoLinkHeader'
+
+type Value = {
+  name: string
+  value: string
+  values?: string[]
+}
+
+type FileList = {
+  [key: string]: {
+    name: string
+    value: string
+    values?: Value[]
+  }
+}
+
+type List = {
+  selector: string
+  property: string
+  token: string
+  value: string
+  values?: string[]
+}
+
+type FilteredRows = {
+  cells: React.ReactNode[]
+  isOpen?: boolean
+  details?: { parent: number; fullWidth: boolean; data: React.ReactNode }
+}
+
+interface CSSTableProps extends React.HTMLProps<HTMLDivElement> {
+  cssPrefix: string
+  hideSelectorColumn?: boolean
+  selector?: string
+  debounceLength?: number
+  autoLinkHeader?: boolean
+}
+
+const isColorRegex = /^(#|rgb)/
+const mappingAsList = (property: string, values: string[]) => (
+  <List isPlain>
+    <ListItem>{property}</ListItem>
+    {values.map((entry: string) => (
+      <ListItem key={entry} icon={<LevelUpAltIcon className="rotate-90-deg" />}>
+        {entry}
+      </ListItem>
+    ))}
+  </List>
+)
+
+const flattenList = (files: FileList[]) => {
+  const list = [] as List[]
+  files.forEach((file) => {
+    Object.entries(file).forEach(([selector, values]) => {
+      if (values !== undefined) {
+        Object.entries(values).forEach(([key, val]) => {
+          if (typeof val === 'object' && val !== null && 'name' in val) {
+            const v = val as unknown as Value
+            list.push({
+              selector,
+              property: v.name,
+              token: key,
+              value: v.value,
+              values: v.values,
+            })
+          }
+        })
+      }
+    })
+  })
+  return list
+}
+
+export const CSSTable: React.FunctionComponent<CSSTableProps> = ({
+  cssPrefix,
+  hideSelectorColumn = false,
+  selector,
+  debounceLength = 500,
+  autoLinkHeader,
+}) => {
+  const prefixToken = cssPrefix.replace('pf-v6-', '').replace(/-+/g, '_')
+
+  const applicableFiles = Object.entries(tokensModule)
```

Comment:
I wonder if we might want to split the actual process of pulling in all the tokens and grabbing the relevant ones into a dedicated API route instead of having it wrapped into the component directly, but that's 100% something we can discuss down the road.

Reply:
If I wanted to know what the best file structure would be for this endpoint, what do you recommend? Do we want an api/[prefix] route or something? This just came up - wondering if I can do this as a quick-follow-on after this merges.

Reply:
Yep it can definitely be a followup 🙂